x/escrow/handler_test.go

```diff
@@ -725,23 +713,29 @@ func (q query) check(t testing.TB, db weave.ReadOnlyKVStore, qr weave.QueryRoute
 	t.Helper()
 
 	h := qr.Handler(q.path)
-	require.NotNil(t, h, q.path)
+	assert.Equal(t, true, h != nil)
 	mods, err := h.Query(db, q.mod, q.data)
 	if q.isError {
-		require.Error(t, err)
+		assert.Equal(t, true, err != nil)
 		return
 	}
-	require.NoError(t, err)
-	if assert.Equal(t, len(q.expected), len(mods), msg...) {
-		for i, ex := range q.expected {
-			// make sure keys match
-			key := q.bucket.DBKey(ex.Key())
-			assert.Equal(t, key, mods[i].Key)
-
-			// parse out value
-			got, err := q.bucket.Parse(nil, mods[i].Value)
-			require.NoError(t, err)
-			assert.EqualValues(t, ex.Value(), got.Value(), msg...)
+	assert.Nil(t, err)
+
+	if len(q.expected) != len(mods) {
+		t.Fatal("Expected modification number is not equal to actual, given following messages", msg...)
+	}
+
+	for i, ex := range q.expected {
+		// make sure keys match
+		key := q.bucket.DBKey(ex.Key())
+		assert.Equal(t, key, mods[i].Key)
+
+		// parse out value
+		got, err := q.bucket.Parse(nil, mods[i].Value)
+		assert.Nil(t, err)
+
+		if !reflect.DeepEqual(ex.Value(), got.Value()) {
```

Comment:
Yes, `assert.Equal` does `reflect.DeepEqual`, but doesn't let us add a message. I see why you did this, and it is correct, but still think it could be cleaner.

How about just using assert.Equal() here and remove `msg..` arg (I think my hack). Rather use 

```go
name := fmt.Sprintf(msg...) // or something cleaner
t.Run(name, func (t *testing.T) {
  check(t, db, qr)
}
```

Then all asserts have context info